Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/DscRepository.kt

```diff
@@ -31,17 +32,24 @@ class DscRepository @Inject constructor(
             replayExpirationMillis = 0
         ),
     ) {
-
-        // TODO
-        DscData(
-            dscList = emptyList()
-        )
+        localStorage.load() ?: defaultDscData.getDscData()
     }
 
     val dscData = internalData.data
 
     suspend fun refresh() {
         Timber.tag(TAG).d("refresh()")
+        internalData.updateBlocking {
+            dscServer.getDscList().let { rawData ->
+                mapDscList(rawData).apply {
+                    localStorage.save(rawData)
+                }
+            }
```

Comment:
what happens if `getDscList` or `mapDscList` throw and error ?

Reply:
It's handled by `DccStateCheckScheduler`
```kotlin
            try {
                    Timber.tag(TAG).i("Refreshing DSC data.")
                    dscRepository.refresh()
                } catch (e: Exception) {
                    Timber.tag(TAG).e(e, "Refreshing DSC data failed.")
                }
```